Fix hp-align crash after printing calibration page and setting calibration
Fix hp-align TypeError and AttributeError
Date: 2024-06-08
Signed-off-by: Zihan Zhu <zhzhu722@reiyy.com>
===================================================================

--- hplip-3.23.12/base/maint.py	2023-11-27 23:06:40.000000000 +0800
+++ /usr/share/hplip/base/maint.py	2024-06-08 20:11:22.738229631 +0800
@@ -684,13 +684,14 @@
     if to_bytes_utf8("manual") in data:
         log.debug("Device supports manual calibration")
         data = status.StatusType10FetchUrl(func, "/Calibration/Session", "<cal:CalibrationState xmlns:cal=\\\"http://www.hp.com/schemas/imaging/con/cnx/markingagentcalibration/2009/04/08\\\" xmlns:dd=\\\"http://www.hp.com/schemas/imaging/con/dictionaries/1.0/\\\">Printing</cal:CalibrationState>")
-        import string
-        data = string.split(data, "/Jobs")[1]
-        data = string.split(data, "\r\n")[0]
+        data = data.decode('utf-8')
+        data = data.split("/Jobs")[1]
+        data = data.split("\r\n")[0]
         data = "/Jobs" + data
         data = status.StatusType10FetchUrl(func, data)
-        data = string.split(data, "Session/")[1]
-        data = string.split(data, "<")[0]
+        data = data.decode('utf-8')
+        data = data.split("Session/")[1]
+        data = data.split("<")[0]
         data = "/Calibration/Session/" + data + "/ManualSelectedPatterns.xml"
         global calibrationSession
         calibrationSession = data
@@ -705,8 +706,8 @@
     if not data:
         return 0
 
-    while "ParmsRequested" not in data:
-        if "CalibrationValid" in data:
+    while b"ParmsRequested" not in data:
+        if b"CalibrationValid" in data:
             return
         data = status.StatusType10FetchUrl(func, "/Calibration/State")
     data = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<!-- THIS DATA SUBJECT TO DISCLAIMER(S) INCLUDED WITH THE PRODUCT OF ORIGIN. -->\n<ManualSelectedPatterns xmlns=\"http://www.hp.com/schemas/imaging/con/cnx/markingagentcalibration/2009/04/08\" xmlns:locid=\"http://www.hp.com/schemas/imaging/con/ledm/localizationids/2007/10/31/\" xmlns:psdyn=\"http://www.hp.com/schemas/imaging/con/ledm/productstatdyn/2007/10/31\"><SelectedPattern><Identifier><Id>1</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern><SelectedPattern><Identifier><Id>2</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern><SelectedPattern><Identifier><Id>3</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern><SelectedPattern><Identifier><Id>4</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern><SelectedPattern><Identifier><Id>5</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern><SelectedPattern><Identifier><Id>6</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern><SelectedPattern><Identifier><Id>7</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern><SelectedPattern><Identifier><Id>8</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern><SelectedPattern><Identifier><Id>9</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern></SelectedPattern><SelectedPattern><Identifier><Id>10</Id></Identifier><Choice><Identifier><Id>%s</Id></Identifier></Choice></SelectedPattern></ManualSelectedPattern>" % ( a, b, c, d, e, f, g, h, i, j )
